CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 3.13.4)
+project(CMaize_Public_Depend)
+
+include("${PROJECT_SOURCE_DIR}/cmake/get_cpp.cmake")
+
+cpp_find_or_build_dependency(
+    CMakePublic
+    URL github.com/ally3999/CMakePublic
+    BUILD_TARGET CMakePublic
+    FIND_TARGET CMakePublic::CMakePublic
+)
+
+cpp_add_library(
+    CMaizePublicDepend
+    SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}"
+    INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}"
+    DEPENDS CMakePublic
+)
+
+target_include_directories (CMaizePublicDepend PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+
```

Comment:
Since CMaize currently doesn't support install does that mean I should use the CMake install as I did in CMakePublic's CMakeLists.txt? Should this also have a config file made for it?

Reply:
It's probably easier to do this in steps. So I'd first do an install similar to CMakePublic. Once we have the whole stack (up to CMakePublicDepend2) working we can then write a `cpp_install` (probably should switch the prefixes on the CMaize functions to `cmaize_` at some point, but that's going to break things) and add it to CMaize. Then we can rewrite this repo, and CMakePublicDepend2, in terms of the `cpp_install` function.

Yeah it's going to need a config file. Any project which may need to potentially be discovered by CMake needs one.